.github/workflows/e2e-versions.yml

```diff
@@ -18,7 +18,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [macos-latest, windows-latest, ubuntu-latest]
-        distribution: ['adopt', 'zulu']
+        distribution: ['adopt', 'adopt-hotspot', 'adopt-openj9', 'zulu']
```

Comment:
Considering the fact that downloading / installation logic is pretty similar for both `adopt-hotspot` and `adopt-openj9`, I don't think that we should include all these use-cases to matrix because it will produce the huge test matrix for thousands of builds that will take much time for running

I suggest
- Add only `adopt-openj9` to `setup-java-major-versions` job. So the final list will be `['adopt', 'adopt-openj9', 'zulu']`
- Don't add anything to other jobs